homeassistant/components/myuplink/sensor.py

```diff
@@ -0,0 +1,162 @@
+) -> None:
+    """Set up myUplink sensor."""
+    entities: list[SensorEntity] = []
+    coordinator: MyUplinkDataCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    # Setup device sensors
+    for device_id in coordinator.data.devices:
+        for description in DEVICE_SENSORS:
+            entities.append(
+                MyUplinkDeviceSensor(
+                    coordinator=coordinator,
+                    device_id=device_id,
+                    entity_description=description,
+                    unique_id_suffix=description.key,
+                ),
+            )
+
+    # Setup device point sensors
+    for device_id, point_data in coordinator.data.points.items():
+        for point_id, device_point in point_data.items():
+            entities.append(
+                MyUplinkDevicePointSensor(
+                    coordinator=coordinator,
+                    device_id=device_id,
+                    device_point=device_point,
+                    entity_description=DEVICE_POINT_DESCRIPTIONS.get(
+                        device_point.parameter_unit
+                    ),
+                    unique_id_suffix=point_id,
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class MyUplinkDeviceSensor(MyUplinkEntity, SensorEntity):
+    """Representation of a myUplink device sensor."""
+
+    entity_description: MyUplinkDeviceSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        device_id: str,
+        entity_description: MyUplinkDeviceSensorEntityDescription,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            device_id=device_id,
+            unique_id_suffix=unique_id_suffix,
+        )
+        self.entity_description = entity_description
+
+    @property
+    def native_value(self) -> StateType:
+        """Sensor state value."""
+        return self.entity_description.value_fn(self.coordinator.data, self.device_id)
+
+
+class MyUplinkDevicePointSensor(MyUplinkEntity, SensorEntity):
+    """Representation of a myUplink device point sensor."""
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        device_id: str,
+        device_point: DevicePoint,
+        entity_description: SensorEntityDescription | None,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            device_id=device_id,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        # Internal properties
+        self.point_id = device_point.parameter_id
+
+        if entity_description is not None:
+            self.entity_description = entity_description
+        else:
+            self._attr_native_unit_of_measurement = device_point.parameter_unit
+            self._attr_name = device_point.parameter_name
```

Comment:
From the screenshot in the PR description it looks like the dame device can have multiple temperature device points. Then we want to distinguish these by the name.

There's currently a PR in progress that allows a placeholder to be added to the entity name. If that gets more eged soon we can use that approach. Otherwise we probably need to override the name even if there's an entity description and use the `parameter_name` as name in all cases.

Reply:
I'm looking a bit more at the screenshot and I don't think we should use placeholders since the parameter names should probably be translated if we want entity name translations.

For now we should probably just set the entity name to the device point parameter name. This won't allow entity name translation but we can look at that in a follow up.

How many different kinds of device point parameters are there?